utils/zerotrie/src/builder/dense.rs

```diff
@@ -28,6 +28,31 @@ pub(crate) struct DenseSparse2dAsciiWithFixedDelimiterBuilder<'a> {
 }
 
 impl<'a> DenseSparse2dAsciiWithFixedDelimiterBuilder<'a> {
+    ///Helper function: finds best row offset when value range too large for dense matrix
+    fn find_window(values: &BTreeMap<&'a str, usize>) -> usize {
+        let mut sorted_vals: Vec<usize> = values.values().cloned().collect();
+        let row_width = usize::from(DenseType::MAX);
+        sorted_vals.sort_unstable();
+        let mut bot = 0;
+        let mut best = 0;
+        let mut best_index = 0;
+        for top in 0..sorted_vals.len() {
+            while bot < top {
+                let top_val = sorted_vals.get(top).copied().unwrap_or(0);
+                let bot_val = sorted_vals.get(bot).copied().unwrap_or(0);
```

Comment:
Issue: Here and everywhere else you changed to use `.unwrap_or(0)`:

It is easy to see that both `top` and `bot` are `< sorted_vales.len()`, so typically our style is to use the index operator and add a suppression with comment like

```
#[allow(clippy::indexing_slicing)] // bot < top and top < sorted_vals.len()
```

We do the `.unwrap_or` style in GIGO situations, where the input _could_ be invalid, but in this case it is an easily provable code constraint.